src/circuit/gadget/ecc/chip/mul/overflow.rs

```diff
@@ -135,16 +135,10 @@ impl Config {
                 self.q_mul_overflow.enable(&mut region, offset + 1)?;
 
                 // Copy `z_0`
-                copy(&mut region, || "copy z_0", self.advices[0], offset, &*zs[0])?;
+                zs[0].copy_advice(|| "copy z_0", &mut region, self.advices[0], offset)?;
 
                 // Copy `z_130`
-                copy(
-                    &mut region,
-                    || "copy z_130",
-                    self.advices[0],
-                    offset + 1,
-                    &*zs[130],
-                )?;
+                zs[130].copy_advice(|| "copy z_130", &mut region, self.advices[0], offset + 1)?;
 
                 // Witness η = inv0(z_130), where inv0(x) = 0 if x = 0, 1/x otherwise
```

Comment:
(For future reference) It's a common pattern to use inv0 to implement nonzero constraints. I'd really like to have an easy way to defer these inverses so that they can be *automatically* batched. To do that, the chip code shouldn't compute the inverse itself, it should call some method on the region that assigns an inverse. Then any use of an inverted value triggers the batch inversion.

Reply:
We already have support for this during witness assignment: that's what `Assigned<F>` represents.

Reply:
So instead of (for example) `z_130.invert().unwrap()` should we do `(pallas::Base::one(), z_130).into()`?

Reply:
`Assigned::from(z_130).invert()` would be clearer.

Reply:
I tried this locally, and discovered a bug in the `AssignedCell` `halo2` PR! Fixed here: https://github.com/zcash/halo2/pull/417 (which I'll immediately start depending on in this PR).

Reply:
Hmm, actually this gets more complex to migrate to in all but a single case, so I'll do this in a separate PR.